### PR TITLE
[macrecovery] Guess downloaded recovery version

### PR DESCRIPTION
**Pull request by dreamwhite**
The aim of this PR is to improve my previous PR #394.
More specifically, with the proposed changes, `macrecovery` can write onto `com.apple.recovery.boot/.contentDetails` the version of the downloaded OS, obtained with the following steps:

1. mount and get the mount point of the downloaded `BaseSystem.dmg` available in `com.apple.recovery.boot` folder (by using `hdiutil mount -plist com.apple.recovery.boot/BaseSystem.dmg`)
2. get the `DTPlatformVersion` in `Install macOS XXXXX.app/Contents/Info.plist` key
3. write onto `com.apple.recovery.boot/.contentDetails` 

Please note that this may slow the execution times of the script itself, mainly due to `hdiutil`. Don't know if there are different ways of mounting a DMG in macOS using the command line, but that's what I used back in the past. I'm opened to suggestions

Kind regards
dreamwhite

### CODE REVIEW

File: Utilities/macrecovery/macrecovery.py [hunk 1]
```diff
@@ -453,6 +455,29 @@ def action_guess(args):
     print(f'UNKNOWN: Failed to determine supported models for MLB {mlb}!')
     return None
 
+def get_basesystem_version(args):
+    print('Extracting recovery version...')
+    os.system(f'hdiutil mount -plist {args.outdir}/BaseSystem.dmg > mount.plist')
```
Comment (williambj1) [inline Utilities/macrecovery/macrecovery.py:460]: This breaks compatibility for Windows and Linux.

Comment (dreamwhite) [reply Utilities/macrecovery/macrecovery.py:460]: O_O I completely forgot about it. Should I use a conditional like `if platform.system() == 'Darwin'` and execute the whole code, or whatever?

**Comment by Andrey1970AppleLife** [comment]
The majority the user use this utility not from macOS.

**Comment by dreamwhite** [comment]
> The majority the user use this utility not from macOS.

Yeah I noticed that. What about adding a condition like `if os is macos, then get the version from the dmg`?

**Pull request closed without merging**